jdaviz/configs/default/plugins/collapse/collapse.py

```diff
@@ -130,35 +132,44 @@ def vue_collapse(self, *args, **kwargs):
         if self.selected_axis == 0:
             spec_min = float(self.spectral_min) * u.Unit(self.spectral_unit)
             spec_max = float(self.spectral_max) * u.Unit(self.spectral_unit)
-            spec = spec.spectral_slab(spec_min, spec_max)
-
-        collapsed_spec = getattr(spec, self.selected_func.lower())(
-            axis=self.selected_axis)
-
-        data = Data(coords=collapsed_spec.wcs)
-        data['flux'] = collapsed_spec.filled_data[...]
+            spec = spectral_slab(spec, spec_min, spec_max)
```

Comment:
This requires resolution of astropy/specutils#909 upstream and then a new `specutils` release including that fix.